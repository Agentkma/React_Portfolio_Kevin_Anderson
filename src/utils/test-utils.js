import "@testing-library/jest-dom";
import userEvent from "@testing-library/user-event";
import { Router } from "react-router-dom";
import { createMemoryHistory } from "history";

const renderWithRouter = (
    ui,
    {
        route = "/",
        history = createMemoryHistory({ initialEntries: [route] }),
    } = {}
) => {
    const Wrapper = ({ children }) => (
        <Router history={history}>{children}</Router>
    );
    return {
        ...render(ui, { wrapper: Wrapper }),
        // adding `history` to the returned utilities to allow us
        // to reference it in our tests (just try to avoid using
        // this to test implementation details).
        history,
    };
};

// re-export everything
export * from "@testing-library/react";

// override render method
export { renderWithRouter, userEvent };
