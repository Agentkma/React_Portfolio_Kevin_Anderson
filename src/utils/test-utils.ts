import "@testing-library/jest-dom";
import userEvent from "@testing-library/user-event";
import { BrowserRouter } from "react-router-dom";
import { render } from "@testing-library/react";
import { ReactNode } from "react";

const renderWithRouter = (ui: React.ReactElement, { route = "/" } = {}) => {
  window.history.pushState({}, `Test page at route ${route}`, route);

  return {
    user: userEvent.setup(),
    ...render(ui, { wrapper: BrowserRouter }),
  };
};

// re-export everything
export * from "@testing-library/react";

// override render method
export { renderWithRouter, userEvent };
