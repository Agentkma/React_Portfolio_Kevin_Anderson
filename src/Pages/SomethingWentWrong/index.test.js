// import dependencies
import React from "react";

import { SomethingWentWrong } from "./index";
// import react-testing methods
import { render } from "test-utils";

test("snapshot", () => {
    const { asFragment } = render(<SomethingWentWrong />);

    expect(asFragment()).toMatchSnapshot();
});
