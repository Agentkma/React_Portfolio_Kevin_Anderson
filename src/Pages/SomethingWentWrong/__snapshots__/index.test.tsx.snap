// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-eOzmre gUQBoy"
  >
    <div
      class="sc-blHHSb sc-eauhAA cYpmAB NIRqt"
    >
      <div
        class="sc-fAUdSK gkZxdF"
      >
        <section
          class="sc-fFoeYl bmMKZz"
        >
          <div
            class="sc-gtLWhw cqYCkV"
          >
            <h3
              class="sc-gQaihK jFMBll"
            >
              Oops!
            </h3>
             
            <div
              class="sc-dntaoT sc-ivxoEo bGSjd jdOLTe"
            />
             
            <h6
              class="sc-iuUfFv myTQq"
            >
              Something Went Wrong!
            </h6>
          </div>
        </section>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
